client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx

```diff
@@ -0,0 +1,98 @@
+import { Button, Card } from '@automattic/components';
+import { getQueryArg } from '@wordpress/url';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { ReactElement, useCallback, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import FormRadio from 'calypso/components/forms/form-radio';
+import SearchCard from 'calypso/components/search-card';
+import { addQueryArgs } from 'calypso/lib/url';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice, infoNotice } from 'calypso/state/notices/actions';
+import useAssignLicenseMutation from 'calypso/state/partner-portal/licenses/hooks/use-assign-license-mutation';
+import './style.scss';
+
+export default function AssignLicenseForm( { sites }: any ): ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const [ filter, setFilter ] = useState( false );
+	const [ selectedSite, setSelectedSite ] = useState( false );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const licenseKey = getQueryArg( window.location.href, 'key' ) as string;
```

Comment:
Not that important: We should follow the convention of page context functions passing query parameters down as component parameters (e.g. `licensesContext`) rather than components getting query parameters on their own, so components are "more" reusable.